presto-oracle/src/main/java/io/prestosql/plugin/oracle/OracleClient.java

```diff
@@ -0,0 +1,215 @@
+import static io.prestosql.plugin.jdbc.StandardColumnMappings.booleanWriteFunction;
+import static io.prestosql.plugin.jdbc.StandardColumnMappings.decimalColumnMapping;
+import static io.prestosql.plugin.jdbc.StandardColumnMappings.integerWriteFunction;
+import static io.prestosql.plugin.jdbc.StandardColumnMappings.smallintColumnMapping;
+import static io.prestosql.plugin.jdbc.StandardColumnMappings.smallintWriteFunction;
+import static io.prestosql.plugin.jdbc.StandardColumnMappings.timestampWriteFunction;
+import static io.prestosql.plugin.jdbc.StandardColumnMappings.tinyintWriteFunction;
+import static io.prestosql.plugin.jdbc.StandardColumnMappings.varcharColumnMapping;
+import static io.prestosql.plugin.jdbc.StandardColumnMappings.varcharWriteFunction;
+import static io.prestosql.spi.StandardErrorCode.NOT_SUPPORTED;
+import static io.prestosql.spi.type.DecimalType.createDecimalType;
+import static io.prestosql.spi.type.VarcharType.createUnboundedVarcharType;
+import static io.prestosql.spi.type.VarcharType.createVarcharType;
+import static io.prestosql.spi.type.Varchars.isVarcharType;
+import static java.lang.String.format;
+import static java.util.Locale.ENGLISH;
+import static java.util.Objects.requireNonNull;
+
+public class OracleClient
+        extends BaseJdbcClient
+{
+    private final boolean synonymsEnabled;
+    private final int varcharMaxSize;
+    private final int timestampDefaultPrecision;
+    private final int numberDefaultScale;
+    private final RoundingMode numberRoundingMode;
+
+    @Inject
+    public OracleClient(
+            BaseJdbcConfig config,
+            OracleConfig oracleConfig,
+            ConnectionFactory connectionFactory)
+    {
+        super(config, "\"", connectionFactory);
+
+        requireNonNull(oracleConfig, "oracle config is null");
+        this.synonymsEnabled = oracleConfig.isSynonymsEnabled();
+        this.varcharMaxSize = oracleConfig.getVarcharMaxSize();
+        this.timestampDefaultPrecision = oracleConfig.getTimestampDefaultPrecision();
+        this.numberDefaultScale = oracleConfig.getNumberDefaultScale();
+        this.numberRoundingMode = oracleConfig.getNumberRoundingMode();
+    }
+
+    private String[] getTableTypes()
+    {
+        if (synonymsEnabled) {
+            return new String[] {"TABLE", "VIEW", "SYNONYM"};
+        }
+        return new String[] {"TABLE", "VIEW"};
+    }
+
+    @Override
+    protected ResultSet getTables(Connection connection, Optional<String> schemaName, Optional<String> tableName)
+            throws SQLException
+    {
+        DatabaseMetaData metadata = connection.getMetaData();
+        String escape = metadata.getSearchStringEscape();
+        return metadata.getTables(
+                connection.getCatalog(),
+                escapeNamePattern(schemaName, escape).orElse(null),
+                escapeNamePattern(tableName, escape).orElse(null),
+                getTableTypes());
+    }
+
```

Comment:
need to set fetch size, default 10 is too slow

Reply:
@ebyhr could we set a default, without be configurable?
or revert the last change

Reply:
configurable pls

Reply:
@eskabetxe We can set default value as RedshiftClient and PostgreSqlClient. Both clients are setting `1000`.